Todo app

voletiswaroopfw opened the pull request.

voletiswaroopfw committed:
Completed app without serializer

voletiswaroopfw committed:
replaced git api with my-json

voletiswaroopfw committed:
refactored code

voletiswaroopfw committed:
updated

umashankar-masilamani-e2221 commented on app/routes/inbox.js:
```diff
@@ -0,0 +1,7 @@
+
+export default Ember.Route.extend({  
+  model: function () {
+    return this.store.findAll('email');
```
lets have not api in the model as they are render blocking

umashankar-masilamani-e2221 commented on app/routes/email-details.js:
```diff
@@ -0,0 +1,7 @@
+
+export default Ember.Route.extend({
+  model: function (params) {
+    return this.store.findRecord("email", params.email_id);
```
same thing here also

umashankar-masilamani-e2221 commented on app/templates/email-details.hbs:
```diff
@@ -0,0 +1,17 @@
+    {{#link-to "index" class="align-center active"}}Home{{/link-to}}
+    {{#link-to "inbox" class="align-center active"}}Inbox{{/link-to}}
+  </nav>
+  <h1>Details page</h1>
```
lets componentise this logic

umashankar-masilamani-e2221 commented on app/templates/settings.hbs:
```diff
@@ -0,0 +1,5 @@
+<nav>
+  {{#link-to "index" class="align-center active"}}Home{{/link-to}}
+  {{#link-to "inbox" class="align-center active"}}Inbox{{/link-to}}
+</nav>
```
lets move nav bar into component and set the active class according to the current route

umashankar-masilamani-e2221 commented on app/components/app-settings/component.js:
```diff
@@ -0,0 +1,30 @@
+const { set, getProperties, setProperties } = Ember;
+
+export default Ember.Component.extend({
+  detailsPage: "",
```
bool

umashankar-masilamani-e2221 commented on app/components/app-settings/component.js:
```diff
@@ -0,0 +1,30 @@
+  init() {
+    this._super(...arguments);
+    let settings = JSON.parse(localStorage.getItem("settings"));
+    setProperties(this, {
```
set

umashankar-masilamani-e2221 commented on app/components/app-settings/component.js:
```diff
@@ -0,0 +1,30 @@
+    },
+
+    saveSettings() {
+      let { detailsPage } = getProperties(this, "detailsPage");
```
get

umashankar-masilamani-e2221 commented on app/components/email-item/component.js:
```diff
@@ -0,0 +1,12 @@
+import Ember from "ember";
+const { set, getProperties, setProperties } = Ember;
+
+export default Ember.Component.extend({
```
export default Ember.Component.extend({
   emailDetailsEnabled: computed({
     get() {
       //fetch from local storage and set it, if not present set it default
    }
  }
})

umashankar-masilamani-e2221 commented on app/components/email-item/template.hbs:
```diff
@@ -0,0 +1,17 @@
+{{#unless emailDetailsEnabled}}
+  <div class="item-desc">
```
lets not duplicate the template in both if and else... lets have the logic of routing to a new route or loading in the right pane inside an action

umashankar-masilamani-e2221 commented on app/components/email-list/template.hbs:
```diff
@@ -0,0 +1,27 @@
+  <ul class="email-list">
+    {{#each emailList as |item|}}
+      <li class="list-item {{item.read}}" onclick={{action "getEmailDetails" item}}>
+        {{email-item item=item}}
```
{{email-item item=item loadDetails=(action 'loadDetails') }}

umashankar-masilamani-e2221 commented on app/components/email-list/component.js:
```diff
@@ -0,0 +1,57 @@
+      set(this, "defaultEmailDetails", selectedEmailDetails);
+    },
+    toggleAll() {
+      let { emailList, markAllItemsRead, toggleAllLabel } = getProperties(
```
logic here should be,
1. lets create a boolean variable isRead: false
2. create a computed property toggleAllLabel -> dependent on isRead boolean variable and return appropriate "Mark as read" or "Mark as unread".
3. create an action toggleEmails()  -> run the loop of all emails and set a toggled value of isRead and finally toggle isRead value


u will get the state of toggleAllLabel which will either be "read" or "unread"

umashankar-masilamani-e2221 commented on app/components/email-list/component.js:
```diff
@@ -0,0 +1,57 @@
+      let finalData = [
+        emailList.filterBy("read", false),
+        emailList.filterBy("read", true),
+      ];
```
for(... run thru emailList) {
   if read, push to readArray
   if unread, push to unreadArray
}
finally concat read and unread

umashankar-masilamani-e2221 replied:
this way we can avoid two loops

umashankar-masilamani-e2221 commented on app/components/header-menu/component.js:
```diff
@@ -0,0 +1,14 @@
+  actions: {
+    toggleMenu() {
+      let {showMenu} = getProperties(this, 'showMenu');
+      set(this, 'activeMenuClass', !showMenu ? 'hamburger-menu menu-activated' : 'hamburger-menu');
+      set(this,'showMenu', !showMenu);
```
setProperties

umashankar-masilamani-e2221 reviewed (changes requested):
first round of review

voletiswaroopfw committed:
Updated code

- Created Nav component
- Utilizing existing details comp for email details route template
- moved set - setProperties and vise-versa
- moved local-storage item from init to computed
- removed un-used component.js files

voletiswaroopfw committed:
revamped showUnreadFirst code

- revamped showUnreadFirst code
- Updated emberJS to 2.9.1

voletiswaroopfw committed:
Updated css for unread items

voletiswaroopfw committed:
Moved routing logic to action

voletiswaroopfw committed:
saving modified data

voletiswaroopfw committed:
Moved api call to components

umashankar-masilamani-e2221 commented on app/components/email-list/component.js:
```diff
@@ -1,57 +1,95 @@
+      if (emailDetails) {
+        this.get("router").transitionTo(`/inbox/email-details/${item.id}`);
+      } else {
+        set(this, "defaultEmailDetails", emailList.filterBy("id", item.id));
```
lets use findBy instead of filterBy and lets call it as selectedEmail

umashankar-masilamani-e2221 commented on app/components/email-list/component.js:
```diff
@@ -1,57 +1,95 @@
+  }).drop(),
+
   actions: {
     getEmailDetails(item) {
```
lets rename this action as it is not a getter

umashankar-masilamani-e2221 commented on app/components/email-list/template.hbs:
```diff
@@ -1,27 +1,26 @@
+      </ul>
+      {{#unless emailDetails.emailDetails}}
+        <div class="email-details">
+          {{#each defaultEmailDetails as |item|}}
```
this shouldn't be an array

umashankar-masilamani-e2221 commented on app/components/email-list/component.js:
```diff
@@ -1,57 +1,95 @@
+            : "email-container",
+      });
+    },
+  }),
```
emailDetails: computed({
get() {
     let settings = JSON.parse(localStorage.getItem("settings"));
    return settings !== null ? settings.detailsPage : false;
}
}

emailWrapperClass: computed({
get() {
let settings = JSON.parse(localStorage.getItem("settings"));
  return settings && settings.detailsPage == true
            ? "email-container email-details-disabled"
            : "email-container";
})

umashankar-masilamani-e2221 commented on app/components/email-item-details/component.js:
```diff
@@ -1,4 +1,25 @@
 export default Ember.Component.extend({
+  init() {
+    this._super(...arguments);
+    if (!this.get("item")) {
```
just like real world scenario, lets add few additional attributes in the email get api and in that scenario we may not be able to consume item property as is.. This details component should always hit an api to get complete detials of a given item.

umashankar-masilamani-e2221 commented on app/components/email-item/component.js:
```diff
@@ -1,12 +1,15 @@
+  emailDetailsEnabled: computed({
+    get() {
+      let settings = JSON.parse(localStorage.getItem("settings"));
+      return set(
```
return settings ? settings.detailsPage : false 
should be good enough

umashankar-masilamani-e2221 reviewed (changes requested):
added second round of comments

voletiswaroopfw committed:
removed set properties from the computed

voletiswaroopfw committed:
updated code
